tests/mr/device/cuda_async_mr_tests.cpp

```diff
@@ -67,6 +68,27 @@ TEST_F(AsyncMRTest, DifferentPoolsUnequal)
   EXPECT_FALSE(mr1.is_equal(mr2));
 }
 
+TEST_F(AsyncMRTest, HWDecompressEnabled)
```

Comment:
Could we isolate this into a separate test executable? I would like to keep the main async MR test from linking to the driver. We want very much to isolate driver linkage and keep it to the minimum possible. If there is any way to do this via the runtime, even if it is a slightly weaker test, that would be preferred. Are there runtime APIs that leverage the decompression engine and would fail cleanly if unavailable?

Please also verify that the changes in this PR run as expected on actual DE hardware if possible. :)

Reply:
I will defer to @eschmidt-nvidia to answer these questions. You certainly can't prod the pointer attributes appropriately with the runtime API.

Reply:
Split out

Reply:
These APIs are only available through the driver.